### ISSUE TITLE
the fine-tune question

### ISSUE THREAD

**Issue by zy199676** [SysCV/sam-hq#82]
Hello, I would like to use your tuning model for tasks related to freespace detection. Currently, there has been no significant improvement in the effectiveness. Can you provide some suggestions for improvement

**Comment by lkeab** [comment]
during testing, have you tried setting both hq_token_only = True and multiask_ Output= False? And can you show your training loss log?

**Comment by zy199676** [comment]
> during testing, have you tried setting both hq_token_only = True and multiask_ Output= False? And can you show your training loss log?
May I ask what the functions of these two are? （Hq_ Token_ Only=True and multimedia_ Output=False）

**Comment by zy199676** [comment]
> during testing, have you tried setting both hq_token_only = True and multiask_ Output= False? And can you show your training loss log?
May I ask what the functions of these two are? （Hq_ Token_ Only=True and multimedia_ Output=False）

**Comment by zy199676** [comment]
> during testing, have you tried setting both hq_token_only = True and multiask_ Output= False? And can you show your training loss log?
The losses are as follows
[ 0/583] eta: 1:29:53 training_loss: 1.3288 (1.3288) loss_dice: 0.6261 (0.6261) loss_mask: 0.7026 (0.7026) time: 9.2506 data: 1.3218 max mem: 18436

[582/583] eta: 0:00:01 training_loss: 0.0516 (0.1067) loss_dice: 0.0221 (0.0495) loss_mask: 0.0285 (0.0572) time: 1.3528 data: 0.0038 max mem: 18856

Total time: 0:13:27 (1.3846 s / it)

Finished epoch: 0

Averaged stats: training_loss: 0.0516 (0.1067) loss_dice: 0.0221 (0.0495) loss_mask: 0.0285 (0.0572)

Validating...

valid_dataloader len: 296

[ 0/296] eta: 0:08:20 val_boundary_iou_0: 0.9320 (0.9320) val_iou_0: 0.9909 (0.9909) time: 1.6906 data: 0.8216 max mem: 18856

[295/296] eta: 0:00:00 val_boundary_iou_0: 0.9067 (0.9145) val_iou_0: 0.9877 (0.9896) time: 0.7223 data: 0.0028 max mem: 18856

Total time: 0:03:37 (0.7357 s / it)

============================

Averaged stats: val_boundary_iou_0: 0.9067 (0.9145) val_iou_0: 0.9877 (0.9896)

come here save at work_dirs/hq_sa/epoch_0.pth

epoch: 1 learning rate: 0.001

[ 0/583] eta: 0:30:18 training_loss: 0.0411 (0.0411) loss_dice: 0.0184 (0.0184) loss_mask: 0.0226 (0.0226) time: 3.1189 data: 1.7543 max mem: 18856

[582/583] eta: 0:00:01 training_loss: 0.0439 (0.0626) loss_dice: 0.0188 (0.0268) loss_mask: 0.0260 (0.0359) time: 1.3495 data: 0.0046 max mem: 18857

Total time: 0:13:08 (1.3527 s / it)

Finished epoch: 1

Averaged stats: training_loss: 0.0439 (0.0626) loss_dice: 0.0188 (0.0268) loss_mask: 0.0260 (0.0359)

Validating...

valid_dataloader len: 296

[ 0/296] eta: 0:08:59 val_boundary_iou_0: 0.9503 (0.9503) val_iou_0: 0.9936 (0.9936) time: 1.8235 data: 1.1054 max mem: 18857

[295/296] eta: 0:00:00 val_boundary_iou_0: 0.8830 (0.9106) val_iou_0: 0.9841 (0.9889) time: 0.7296 data: 0.0038 max mem: 18857

Total time: 0:03:38 (0.7368 s / it)

============================

Averaged stats: val_boundary_iou_0: 0.8830 (0.9106) val_iou_0: 0.9841 (0.9889)

come here save at work_dirs/hq_sa/epoch_1.pth

epoch: 2 learning rate: 0.001

[ 0/583] eta: 0:29:49 training_loss: 0.0895 (0.0895) loss_dice: 0.0311 (0.0311) loss_mask: 0.0584 (0.0584) time: 3.0699 data: 1.3855 max mem: 18857

[582/583] eta: 0:00:01 training_loss: 0.0355 (0.0557) loss_dice: 0.0153 (0.0238) loss_mask: 0.0199 (0.0319) time: 1.3455 data: 0.0035 max mem: 18857

Total time: 0:13:06 (1.3494 s / it)

Finished epoch: 2

Averaged stats: training_loss: 0.0355 (0.0557) loss_dice: 0.0153 (0.0238) loss_mask: 0.0199 (0.0319)

Validating...

valid_dataloader len: 296

[ 0/296] eta: 0:09:18 val_boundary_iou_0: 0.9498 (0.9498) val_iou_0: 0.9939 (0.9939) time: 1.8881 data: 1.1689 max mem: 18857

[295/296] eta: 0:00:00 val_boundary_iou_0: 0.9122 (0.9223) val_iou_0: 0.9887 (0.9911) time: 0.7267 data: 0.0026 max mem: 18857

Total time: 0:03:36 (0.7331 s / it)

============================

Averaged stats: val_boundary_iou_0: 0.9122 (0.9223) val_iou_0: 0.9887 (0.9911)

come here save at work_dirs/hq_sa/epoch_2.pth

**Comment by lastwhisper-ghy** [comment]
Hi, thanks for your work!
I train the SAM_HQ on my custom dataset,but the val_iou and the val_boundary_iou is very low .for example:
Averaged stats: val_iou_0: 0.0183 (0.2024) val_boundary_iou_0: 0.0183 (0.1962)

Can you give me some tips?
